### ISSUE TITLE
Add option for "simple" Full Screen in OS X

### ISSUE THREAD

**Issue by quinn** [atom/atom#12084]
all tools that developers use provide this option (item2, sublime text). OS X spaces feature is obnoxious. 

https://sublimetext.userecho.com/topics/2372-make-a-fullscreen-mode-without-using-the-os-x-lion-full-screen-junk/

**Comment by quinn** [comment]
and for iTerm: 

https://gitlab.com/gnachman/iterm2/issues/1913

it looks like the iTerm project resolved this issue in 1 day?

**Comment by mnquintana** [comment]
Thanks for the suggestion! This looks like a duplicate of https://github.com/atom/atom/issues/967 – feel free to subscribe there for updates.

**Comment by quinn** [comment]
@mnquintana thanks. the other ticket says "help-wanted", but the discussion is locked. I'd like to try this, where does the currently live? is it part of atom or part of the electron project?

**Comment by callodacity** [comment]
@quinn I'm definitely down to help, did you find anything useful ? 
I found the code that MacVim uses to implements this feature [here](https://github.com/b4winckler/macvim/blob/master/src/MacVim/MMFullScreenWindow.m).

**Comment by quinn** [comment]
@callodacity I also no zero objective C, but thanks to comments this seems to be the relevant lines:

```obj-c

#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5)
    // HACK! Put window on all Spaces to avoid Spaces (available on OS X 10.5
    // and later) from moving the full-screen window to a separate Space from
    // the one the decorated window is occupying.  The collection behavior is
    // restored further down.
    NSWindowCollectionBehavior wcb = [self collectionBehavior];
    [self setCollectionBehavior:NSWindowCollectionBehaviorCanJoinAllSpaces];
#endif

    // make us visible and target invisible
    [target orderOut:self];
    [self makeKeyAndOrderFront:self];

#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5)
    // Restore collection behavior (see hack above).
    [self setCollectionBehavior:wcb];
#endif
```

in the macvim code `self` is the window, and in the electron code `_window` appears to be the window object. I gather this because the both do `setReleasedWhenClosed:NO` and other similar calls to prep the window to enter full screen. It also seems like macvim is doing a bit more stuff to format the window for fullscreen which I assume is necessary if not using the native fullscreen.
